### PR TITLE
feat(app): LPC add pick up tip confirmation 

### LINKED ISSUE #8533: Labware Position Check: add pick up tip modal
# Overview

When moving from checking tipracks to checking labware, the user will need to confirm that they successfully picked up a tip

see [this figma node](https://www.figma.com/file/JCQNOTTK9tTmYPhRqeOfSj/Milestone-0:-Protocol-Upload-Revamp?node-id=4213%3A48721)
# Implementation details

- Show modal in the figma node after a user presses the `Pickup Tip` CTA 
- Main CTA should show the next `moveToWell` command's labware's slot so the user knows where the pipette will move next
- When pressing the CTA in the modal, call the callback exposed by https://github.com/Opentrons/opentrons/issues/8534 to initiate side effects

# Design
https://www.figma.com/file/JCQNOTTK9tTmYPhRqeOfSj/Milestone-0:-Protocol-Upload-Revamp?node-id=4213%3A48721

# Acceptance criteria

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview
This PR adds a pick up tip confirmation modal after a pickup tip command is executed. It drops the tip and resets the state if the user says the pick up tip was unsuccessful, and proceeds with the rest of LPC the user says it was successful. 

This was tested on @Kadee80's robot and works as expected!

Note: The button colors on the pick up tip modal are gray, they need to be changed to match the designs. I'll address this in a follow up after the beta release.

closes #8533

# Changelog

- LPC add pick up tip confirmation 

# Risk assessment
Low

### CODE REVIEW

**Comment by shlokamin** [comment]
oop looks like ive got merge conflicts, code review would still be appreciated to get this in ASAP :) 

i'll be testing on jethary's robot a little later today

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const LabwareOffsetsSummary = (
 ): JSX.Element | null => {
   const { offsetData } = props
   const { t } = useTranslation('labware_position_check')
-
+  console.log(offsetData)
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx]: there's a forgotten `console.log` here

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx [hunk 2]
```diff
@@ -130,19 +130,19 @@ export const LabwareOffsetsSummary = (
                       X
                     </Text>
                     <Text key={x} as={'span'} marginRight={SPACING_2}>
-                      {x.toPrecision(1)}
+                      {x.toFixed(2)}
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx:132]: curious why you changed it from `toPrecision` to `toFixed`?

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/__tests__/LabwarePositionCheck.test.tsx [hunk 3]
```diff
@@ -115,13 +124,37 @@ describe('LabwarePositionCheck', () => {
...
-  // TODO: fix after wiring up
-  it.todo('renders LabwarePositionCheck with Summary Screen component')
-  // TODO: fix after wiring up
-  it.todo(
-    'renders LabwarePositionCheck with Robot Motion Loading Modal component'
-  )
-
-  // TODO: IMMEDIATELY fix this when LabwarePositionCheck/index is final and the isComplete boolean is final
-  it.todo('renders LabwarePositionCheck with IntroScreen component')
+  it('renders the loading screen', () => {
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/__tests__/LabwarePositionCheck.test.tsx:126]: hooray that everything is wired up 🎆

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/styles.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.confirm_pickup_tip_button {
```
Comment (jerader) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/styles.css:1]: how come we want to use a styles sheet instead of inline css?

**Review by jerader — APPROVED** [review]
Things are looking good to me, left a few comments.

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/index.tsx [hunk 5]
```diff
@@ -109,6 +114,54 @@ export const LabwarePositionCheck = (
...
+          title: t('labware_position_check_title'),
+          back: {
+            onClick: confirmExitLPC,
+            title: t('shared:exit'),
+            children: t('shared:exit'),
+          },
+        }}
+      >
+        <SummaryScreen savePositionCommandData={savePositionCommandData} />
+      </ModalPage>
```
Comment (shlokamin) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/index.tsx:143]: all of these calls to `<ModalPage />` can be DRY'd up, there are three instances that are pretty much all copied, so we can make a `LPCModalPage` wrapper or something. I'll add a todo

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/ConfirmPickUpTipModal.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      buttons={[
+        { children: props.denyText, onClick: props.onDeny },
+        {
+          children: props.confirmText,
+          onClick: props.onConfirm,
+          className: styles.confirm_pickup_tip_button,
+        },
+      ]}
+      alertOverlay
+    >
```
Comment (shlokamin) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/ConfirmPickUpTipModal.tsx:25]: these button colors need to be updated to be blue and white, will address this in a follow up

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx (hunk 2 above)
Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwarePositionCheck/LabwareOffsetsSummary.tsx:132]: `toPrecision` was using exponential notation which was unnecessary and confusing to read

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/styles.css (hunk 4 above)
Comment (shlokamin) [reply app/src/organisms/ProtocolSetup/LabwarePositionCheck/styles.css:1]: bcuz `AlertModal` is an older component that doesn't utilize styled components, so we have to pass in styles using css modules

**Review by b-cooper — APPROVED** [review]
Code looks good! Todo comment callouts look good 👍

**Pull request merged**